cmd/cli/app/artifact/artifact_get.go

```diff
@@ -17,60 +17,71 @@ package artifact
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"os"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	"google.golang.org/grpc"
 
+	"github.com/stacklok/minder/cmd/cli/app"
 	"github.com/stacklok/minder/internal/util"
 	"github.com/stacklok/minder/internal/util/cli"
-	pb "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
 )
 
-var artifact_getCmd = &cobra.Command{
+var getCmd = &cobra.Command{
 	Use:   "get",
 	Short: "Get artifact details",
-	Long:  `Artifact get will get artifact details from an artifact, for a given ID`,
-	RunE: cli.GRPCClientWrapRunE(func(ctx context.Context, cmd *cobra.Command, conn *grpc.ClientConn) error {
-		tag := util.GetConfigValue(viper.GetViper(), "tag", "tag", cmd, "").(string)
-		artifactID := viper.GetString("id")
-		latest_versions := viper.GetInt32("latest-versions")
+	Long:  `The artifact get subcommand will get artifact details from an artifact, for a given ID.`,
+	RunE:  cli.GRPCClientWrapRunE(getCommand),
+}
 
-		// tag and latest versions cannot be set at same time
-		if tag != "" && latest_versions != 1 {
-			return errors.New("tag and latest versions cannot be set at the same time")
-		}
+// getCommand is the artifact get subcommand
+func getCommand(ctx context.Context, cmd *cobra.Command, conn *grpc.ClientConn) error {
+	client := minderv1.NewArtifactServiceClient(conn)
 
-		client := pb.NewArtifactServiceClient(conn)
+	provider := viper.GetString("provider")
+	project := viper.GetString("project")
+	tag := viper.GetString("tag")
+	artifactID := viper.GetString("id")
+	latestVersions := viper.GetInt32("versions")
 
-		// check artifact by name
-		art, err := client.GetArtifactById(ctx, &pb.GetArtifactByIdRequest{
-			Id:             artifactID,
-			LatestVersions: latest_versions,
-			Tag:            tag,
-		})
-		if err != nil {
-			return cli.MessageAndError(cmd, "Error getting artifact by id", err)
-		}
+	// Ensure provider is supported
+	if !app.IsProviderSupported(provider) {
```

Comment:
I would actually like to get rid of these checks in the CLI. We will support other providers in the future, and we should just output a relevant error message if the provider isn't registered in the db. So it should be a server-side check instead.